dmutils/email/dm_mandrill.py

```diff
@@ -158,21 +157,21 @@ def decrypt_data(encrypted_data, secret_key, namespace, max_age_in_seconds):
     return json.loads(data.decode('utf-8')), timestamp
 
 
-def _parse_fernet_timestamp(ciphertext):
+def decode_token(token, secret_key, namespace, max_age_in_seconds=86400):
     """
-    Returns utc timestamp embedded in Fernet-encrypted ciphertext, converted to Python datetime object.
+    Decode a token given a secret_key and namespace.
 
-    Decryption should be attempted before using this function, as that does cryptographically strong tests on the
-    validity of the ciphertext.
-    """
-    decoded = base64.urlsafe_b64decode(ciphertext)
+    The token may have been created by a previous version of dmutils, which only supported signed unencrypted tokens.
+    To maintain backwards compatibility during rollouts, we try and decode using the old format, and if that fails
+    assume it is is a new encrypted token.
 
+    This functionality can be removed once all rollouts have completed and the longest time-to-live of the old signed
```

Comment:
The old tokens are long gone, so maybe now is the time to remove this functionality, rather than move it round?